### PR TITLE
Reliability logs

### PR DESCRIPTION
**Pull request by kaadhina**
Task 573918 PS script changes to catch reliability issues
Validations: Manual, Ran UTs.
Compat testing in progress

### CODE REVIEW

**Comment by SatishThareja** [comment]
Looks fine.
Can you mention the testing done and attach a sample log output for any error scenario.

File: Tasks/VsTest/VSTest.ps1 [hunk 1]
```diff
@@ -37,133 +37,141 @@ import-module "Microsoft.TeamFoundation.DistributedTask.Task.TestResults"
...
-            else
-            {
-                if(!$publishResultsOption)
-                {
-                    Write-Warning (Get-LocalizedString -Key "Update the agent to try out the '{0}' feature." -ArgumentList "opt in/out of publishing test run attachments")
-                }
-                Publish-TestResults -Context $distributedTaskContext -TestResultsFiles $resultFiles -TestRunner "VSTest" -Platform $platform -Configuration $configuration
-            }		
+            Write-Host "##vso[task.logissue type=warning;code=002003;]"
+            Write-Warning (Get-LocalizedString -Key "No results found to publish.")
```
Comment (vimegh) [inline Tasks/VsTest/VSTest.ps1]: Can we modify this warning to provide:  $testResultsDirectory & specify that it will search for *.trx

**Comment by vimegh** [comment]
Commented. Looks fine.  +1 to Satish please mention if any testing was done.

File: Tasks/VsTest/VSTest.ps1 [hunk 2]
```diff
@@ -37,133 +37,156 @@ import-module "Microsoft.TeamFoundation.DistributedTask.Task.TestResults"
...
-$sourcesDirectory = Get-TaskVariable -Context $distributedTaskContext -Name "Build.SourcesDirectory"
-if(!$sourcesDirectory)
-{
-    # For RM, look for the test assemblies under the release directory.
-    $sourcesDirectory = Get-TaskVariable -Context $distributedTaskContext -Name "Agent.ReleaseDirectory"
-}
+Write-Host "##vso[task.logissue type=warning;TaskName=VSTest]"
 
-if(!$sourcesDirectory)
+try
```
Comment (acesiddhu) [inline Tasks/VsTest/VSTest.ps1]: can u write a comment why try catch is required

**Comment by acesiddhu** [comment]
rest looks fine

**Pull request merged**